### PR TITLE
Update SIS implementation

### PR DESCRIPTION
**Pull request by claudiofantacci**
This PR updates the implementation of the `SIS` particle filter.
In particular, the `ParticleFilter` class constructor now requires initialization, prediction, correction and resampling classes. Consequently, the `SIS` constructor has been updated and a test on the `ParticleFilter` class deleted.
Other commits clean up code and update the `CHANGELOG`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 ##### `Filtering Algorithms`
```
Comment (xEnVrE) [inline CHANGELOG.md]: Since the interface of class `ParticleFilter` heavily changed, maybe it is better to explicitly indicate that some methods were removed and the constructor has changed. What do you think?

File: CHANGELOG.md [hunk 2]
```diff
@@ -117,13 +119,12 @@
...
 
-## Version 0.6.2.0
-### Updates
+## 🔖 Version 0.6.2.0
 ##### `Filtering classes`
  - Added ResamplingWithPrior class.
  - Added HistoryBuffer and EstimatesExtraction classes.
 
-##### `Filtering Features`
+##### `Filtering features`
```
Comment (xEnVrE) [inline CHANGELOG.md:128]: Since in this commit we are cleaning a bit the Changelog, can we also change `Filtering Functions` to `Filtering functions` and the same for `Algorithms` and `Utilities` in order to enforce this convention?

File: test/test_UKF/main.cpp [hunk 3]
```diff
@@ -95,7 +95,7 @@ int main()
 
     /* Step 2.2 - Define the prediction step. */
 
-    /* Initialize the particle filter prediction step and pass the ownership of the state model. */    
+    /* Initialize the particle filter prediction step and pass the ownership of the state model. */
```
Comment (xEnVrE) [inline test/test_UKF/main.cpp]: I think this is my fault, I wrote this test  and this comment is wrong since the `UKF` has nothing to do with particles. :laughing: 

Can we change this to `Initialize the unscented Kalman filter prediction step and pass the ownership of the state model`. Probably it is better to do this in another commit since unrelated to the description of this one. What do you think?

**Review by xEnVrE — CHANGES_REQUESTED** [review]
LGTM! Thanks! :tada: 

Just some minor concerns regarding the Changelog and one comment.

File: test/test_UKF/main.cpp (hunk 3 above)
Comment (claudiofantacci) [reply test/test_UKF/main.cpp]: Done!

Comment (xEnVrE) [reply test/test_UKF/main.cpp]: Thanks :+1:

File: CHANGELOG.md (hunk 2 above)
Comment (claudiofantacci) [reply CHANGELOG.md:128]: What do you mean? You mean change the capital letter?

Comment (xEnVrE) [reply CHANGELOG.md:128]: Yes, the capital letter of the second word.

Comment (claudiofantacci) [reply CHANGELOG.md:128]: Done!

File: CHANGELOG.md (hunk 1 above)
Comment (claudiofantacci) [reply CHANGELOG.md]: Done!

**Comment by claudiofantacci** [comment]
Pushed new commits.
We should also devise a way to disable logs (write to files, in particular) during tests by default, with the possibility of activating them with a parameter or CMake option.
It happens often that CI build fails because of the logs (time, sizes, etc.).

**Comment by xEnVrE** [comment]
> We should also devise a way to disable logs (write to files, in particular) during tests by default, with the possibility of activating them with a parameter or CMake option.
It happens often that CI build fails because of the logs (time, sizes, etc.)

I agree. What about reducing also the number of particles? If I am not wrong, I launched the tests once without logging and the situation did not change too much (as regards times). I will verify again and let you know.

**Review by xEnVrE — APPROVED** [review]

**Comment by claudiofantacci** [comment]
> I agree. What about reducing also the number of particles? If I am not wrong, I launched the tests once without logging and the situation did not change too much (as regards times). I will verify again and let you know.

Yes, sure can do that. There is no point in using _so many_ particles. I'll open an issue with this points.
